x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java

```diff
@@ -327,6 +376,12 @@ public boolean equals(Object obj) {
 
         @Override
         public Query wildcardQuery(String wildcardPattern, RewriteMethod method, QueryShardContext context) {
+            try {
+                wildcardPattern = normalize(wildcardPattern);
```

Comment:
We should normalize all term-based query like ` KeywordFieldType`.
Also @cbuescher added the support in [`StringFieldType`](https://github.com/elastic/elasticsearch/pull/53512) to mask special characters (`*`)  during normalization so we should reuse here ?